src/libYARP_serversql/src/yarpserver.cpp

```diff
@@ -291,11 +274,58 @@ yarpserversql_API int yarpserver_main(int argc, char *argv[]) {
            nc.where().getHost().c_str(), nc.where().getPort());
     fprintf(out, "\nOk.  Ready!\n");
 
-    while (true) {
-        SystemClock::delaySystem(600);
-        fprintf(out, "Name server running happily\n");
+    while (true)
+    {
+        messageCounter += pollingRate;
+        SystemClock::delaySystem(pollingRate);
+        double dummy;
+
+        if(modf(messageCounter / 600.0, &dummy) < .00001)
+        {
+            fprintf(out, "Name server running happily\n");
+        }
+
+        if(isStopping())
+        {
+            fprintf(out, "closing yarp server\n");
+            break;
+        }
     }
+
     server.close();
+    return;
+}
 
+yarpserversql_API yarp::os::NameStore *yarpserver_create(yarp::os::Searchable& options) {
+    NameServerContainer *nc = new NameServerContainer;
+    if (!nc) return nullptr;
+    nc->setSilent(true);
+    if (!nc->open(options)) {
+        delete nc;
+        return nullptr;
+    }
+    nc->goPublic();
+    return nc;
+}
+
+namespace
+{
+    bool should_exit(false);
+}
+
+yarpserversql_API int yarpserver_main(int argc, char *argv[])
+{
+
+    Network          yarp(yarp::os::YARP_CLOCK_SYSTEM);
+    YarpServerThread yServer;
+
+    std::signal(SIGINT, [](int sig){should_exit = true;});
+    yServer.configure(argc, argv);
+    yServer.start();
+    while(!should_exit && !yServer.isStopping())
+    {
+        yarp::os::Time::delay(0.1);
+    }
+    yServer.stop();
     return 0;
```

Comment:
If I understand the code correctly, there is a regression, as the yarpserver will now always return 0 regardless if it started successfully or not. See https://github.com/robotology/yarp/issues/395 .